### DISCUSSION TITLE
Adding CSS Color Style Information for each Color in Docs

### DISCUSSION THREAD

**Discussion by jwatts777** [saadeghi/daisyui#1658] [Ideas / feature requests]
As the title says. 

Willing to work on it if need be.

Ex. Below

![Untitledzzzzz](https://user-images.githubusercontent.com/91865823/222890548-e6eb7b7b-cc02-4cb1-9b9f-2abd45a20f54.png)

**Comment by amirsinaa** [comment]
I don't think this will add value to the document, but if it is going to be added, I think it is better to use ‍`rgb/rgba` format instead of `HEX`

**Reply by jwatts777** [reply]
Solid point, still 100% usable, and easily transferable to HEX if need be. I slightly edited the post to reflect this.
Thanks!

**Reply by saadeghi** [reply]
The color value is coming from the theme. It can be a built-in theme or a custom theme. So I don't think showing the color value will be helpful. Let me know if there's a reason if it can be helpful 🤔

**Reply by jwatts777** [reply]
My headache that spawned this was trying to get the correct shade of yellow that is used in "retro", and use it in an TS particles component.

If using a DaisyUI theme, it makes sense (atleast to me) to check the docs with that theme enabled, and see a quick reference to use that color, either in rgb/a or as hex. 

I'm not saying if a visitor creates a custom theme, they should see those custom colors but for the baseline themes such as "aqua" and "valentine" that are in-built and beginner friendly, this seems like a missing piece.

**Reply by saadeghi** [reply]
I would suggest to use the CSS variable if you can. For example if you want to use the primary color in JS, you can use `hsl(var(--p))` and it gives you the color value of the primary color from the current theme.  

It's also possible to see the HSL value of a color from the current theme in browser dev tools.

**Reply by amirsinaa** [reply]
> My headache that spawned this was trying to get the correct shade of yellow that is used in "retro", and use it in an TS particles component.
> 
> If using a DaisyUI theme, it makes sense (atleast to me) to check the docs with that theme enabled, and see a quick reference to use that color, either in rgb/a or as hex.
> 
> I'm not saying if a visitor creates a custom theme, they should see those custom colors but for the baseline themes such as "aqua" and "valentine" that are in-built and beginner friendly, this seems like a missing piece.

I first thought your proposal was only for documentation's sake as Tailwind has mentioned the `rgb(a)` equivalent to color classes in its document too.
But as @saadeghi said for your purpose it's best to use a base CSS color variable and play with its `hue`, `saturation`, and `lightness` with HSL color function and you will get a much better and more exact result color

**Reply by jwatts777** [reply]
> I first thought your proposal was only for documentation's sake as Tailwind has mentioned the `rgb(a)` equivalent to color classes in its document too.
> But as @saadeghi said for your purpose it's best to use a base CSS color variable and play with its `hue`, `saturation`, and `lightness` with HSL color function and you will get a much better and more exact result color

My proposal was for exactly that. A reference to the built-in colors for quick copy and paste in the documentation. A la Tailwindcss.com

> I would suggest to use the CSS variable if you can. For example if you want to use the primary color in JS, you can use `hsl(var(--p))` and it gives you the color value of the primary color from the current theme.  
> 
> It's also possible to see the HSL value of a color from the current theme in browser dev tools.  
> 

Will follow this advice
